04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,97 @@
+const Product = require('../models/product')
+const queryValidator = require('../utils/queryValidator')
+
+const DEFAULT_LIMIT = 10;
+const START_PAGE = 1;
+
+const getAllProductsStatic = async (req, res) => {
+  const products = await Product.find(req.query)
+  res.status(200).json({
+    length: products.length,
+    ...req.query,
+    products
+  })
+}
+const getAllProducts = async (req, res) => {
+  const {
+    name,
+    featured,
+    company,
+    sort,
+    fields,
+    numericFilters,
+  } = req.query
+  let {
+    limit = DEFAULT_LIMIT,
+    page  = START_PAGE
+  } = req.query
+  const queryObject = {}
+
+  if (name) {
+    queryObject.name = { $regex: name, $options: 'i' }
+  }
+  if (featured) {
```

Comment:
(optional):
it may improve the user experience to make it case insensitive i.e. so that `true`, `TRUE`, `false`, and `FALSE` are all valid values